src/posts/posts.service.ts

```diff
@@ -92,11 +99,13 @@ export class PostsService {
         }
     
         const isArchived = archived === 'true';
-    
-        return this.prisma.posts.update({
+        
+        const updatedPost = await this.prisma.posts.update({
             where: { id },
             data: { archived: isArchived },
         });
+        const action = await this.actionsService.addAction('Update',userId,'Post',post.id,updatedPost)
```

Comment:
here you have hadcoded strings. Store them in variables and reuse? insted of hardcode them